Modular resource updates for python (#1419)
viamrobotics/docs: docs/extend/modular-resources/_index.md

## Use a modular resource with your robot

> To add a modular resource to your robot configuration, you will sometimes use a modular resource that already exists (that someone else created).
> If this is the case, skip to [Save the executable](#make-sure-viam-server-can-access-your-executable)
> If you are creating your own modular resource, follow these 5 steps:

Reviewer: ```suggestion
If you are using an existing modular resource, you can skip to [Save the executable](#make-sure-viam-server-can-access-your-executable).
If you are creating your own modular resource, follow these steps:
```
- avoid future tense where possible
- shorten (for clarity)
- if it's important who created it we can add "that someone else created" back in
- where possible avoid specifying the number of steps. Easy to forget to update if the number changes.

        return self.left.is_powered(extra=extra, timeout=timeout)[0] or self.right.is_powered(extra=extra, timeout=timeout)[0]
```

> <file>__init.py__</file>

Author: ```suggestion
<file>\__init.py__</file>
```

> <file>\__init.py__</file>

Author: ```suggestion
<file>`__init.py__`</file>
```

Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new_base, MyBase.validate_config))
```

<file>main.py</file>

> ``` python {class="line-numbers linkable-line-numbers"}

Author: ```suggestion

``` python {class="line-numbers linkable-line-numbers"}
```

sudo chmod +x <FILEPATH>/<FILENAME>
```

You need to ensure any dependencies for your module (including the Viam SDK) are installed, as well.

> Your executable will be run by *viam-server* as root, so dependencies will need to be available to the root user.

Reviewer: ```suggestion
Your executable will be run by `viam-server` as root, so dependencies need to be available to the root user.
```
